packages/chatbot-eval-mongodb-public/src/braintrust/evaluateRagConversationsReferenceFree.ts

```diff
@@ -0,0 +1,111 @@
+import * as braintrust from "braintrust";
+import {
+  AnswerRelevancy,
+  ContextRelevancy,
+  Faithfulness,
+  Scorer,
+} from "autoevals";
+import { ConversationGeneratedData } from "mongodb-chatbot-evaluation";
+import { PromisePool } from "@supercharge/promise-pool";
+import { extractConversationEvalData } from "./utils";
+export interface EvaluateRagConversationsParams {
+  projectName: string;
+  conversationGeneratedData: ConversationGeneratedData[];
+  evaluatorConfig: {
+    apiKey: string;
+    modelName: string;
+    baseUrl?: string;
+    headers?: Record<string, string>;
+  };
+  metadata?: Record<string, unknown>;
+  description?: string;
+  experimentName?: string;
+}
+
+const retrievedContext: Scorer<string, { context: string[] }> = async ({
+  context,
+}) => {
+  return {
+    name: "RetrievedContext",
+    score: context.length ? 1 : 0,
+  };
+};
+
+const allowedQuery: Scorer<string, { allowedQuery: boolean }> = async ({
+  allowedQuery,
+}) => {
+  return {
+    name: "AllowedQuery",
+    score: allowedQuery ? 1 : 0,
+  };
+};
+
+/**
+  Evaluates a {@link ConversationGeneratedData} using [Braintrust](https://braintrustdata.com)
+  based on the reference-free evaluation metrics from the `autoevals` library:
+  - {@link Faithfulness}
+  - {@link AnswerRelevancy}
+  - {@link ContextRelevancy}
+
+  Also uses the {@link retrievedContext} metric to check if any context was retrieved.
+ */
+export async function evaluateRagConversationsReferenceFree({
```

Comment:
Do we need tests for this and other evaluators?

Reply:
when we properly integrate braintrust into the framework, will do